### ISSUE TITLE
false positive on 'no-useless-call' when `apply`ing local function on local object for leveraging array spread

### ISSUE THREAD

**Issue by craPkit** [eslint/eslint#9974]
**Tell us about your environment**
macOS 10.13.3 (High Sierra)
Visual Studio Code
Project-local eslint installation

* **ESLint Version:**
Tested on 4.15.0 and 4.17.0 with the same result

* **Node Version:**
8.4.0

* **npm Version:**
5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
none (default!?)

<details>
<summary>Configuration</summary>

```js
"extends": "eslint:recommended",
"env": {
    "browser": true,
    "es6": true
},
"plugins": ["html"],
"rules": {
    "prefer-spread": 0,
    "no-useless-call": 2
}
```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
Context: A Polymer (1.x) web component. This forces me to use the locally defined & bound `splice` method, as opposed to that from Array's prototype.
I need to splice an arbitrary number of `children`, thus I need to `apply` it with a spread array—directly calling `this.splice(...)` is not possible with an unknown number of subjects.

**Code & eslint-command**

```js
this.splice.apply(this, ['items', this.items.length, 0, ...children]);
```

```bash
eslint ./my-local-file.html
```

**What did you expect to happen?**
'no-useless-call' error not triggered due to combination of `apply`/`call` and ES6 array spread `...` in parameters array. 

**What actually happened? Please include the actual, raw output from ESLint.**
Output: `error    unnecessary '.apply()'                    no-useless-call`

**Comment by mysticatea** [comment]
It looks to be working correctly.

You can write `this.splice('items', this.items.length, 0, ...children);`

**Comment by craPkit** [comment]
Gosh! *facepalm*
Sorry for bothering you. And thanks!

**Comment by mysticatea** [comment]
Not at all.
Closing as answered.
